### DISCUSSION TITLE
edit a certificate (strictly for testing purposes)

### DISCUSSION THREAD

**Discussion by ousia** [openssl/openssl#26168] [Q&A]
I may generate a self-signed RSA certificate with:

``` bash
openssl req -x509 -newkey rsa -keyout key.pem -out cert.pem -days 365
```

After creating `cert.pem`, I would like to change its `CN` from `Just Testing` to `Just Nesting`.

I know this will invalidate the signature in the certificate, but I want that to test a program.

Is there any way to change `CN`in `cert.pem` and without replacing its signature or the public key?

Of course, I know this will break the certificate, but this is exactly what I’m aiming at.

Many thanks in advance.

**Comment by beldmit** [comment]
If you look at the cert in the DER format, you will see that CN is represented as a string. So you can edit it either with hex editor or with regexp (taking into account limitations of the regexp engine with arbitrary symbols)
